src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -67,7 +67,7 @@ internal abstract class ODataWriterCore : ODataWriter, IODataOutputInStreamError
 
             this.outputContext = outputContext;
             this.writingResourceSet = writingResourceSet;
-            this.writingDelta = writingDelta;
+            this.writingDelta = writingDelta || (!writingResourceSet && !outputContext.WritingResponse);

```

Comment:
>(!writingResourceSet && !outputContext.WritingResponse) [](start = 48, length = 55)

Can u illustrate the change here?

Reply:
delta is valid when writing any top-level resource in a request.

---
In reply to: [348131350](https://github.com/OData/odata.net/pull/1585#discussion_r348131350) [](ancestors = 348131350)